jsonschema-core.md

```diff
@@ -880,29 +882,50 @@ vocabulary's keywords.
 Meta-schema authoring is an advanced usage of JSON Schema, so the design of
 meta-schema features emphasizes flexibility over simplicity.
 
+#### Dialect Determination
+
+When evaluation encounters a new schema resource (i.e. the lexical scope
+changes), the first task is to determine the dialect used by the schema. To do
+this, implementations MUST determine the dialect using the following prioritized
+steps.
+
+1. The `$schema` keyword - Implementations MUST process the schema according to
+   the dialect it declares.
+2. `application/schema+json` media type with a `schema` parameter -
+   Implementations which support media type parameter inputs MUST process the
+   schema according to the dialect the parameter declares. A media type will
+   generally only be available if the schema has been retrieved from an external
+   source and only applies to the document root.
```

Comment:
I think we can go stronger here and say that if the schema is fetched over HTTP, it MUST respect the media type parameter if it is present, and add a reference to the section that defines the name and syntax of this header. This may be implied by the existing wording, but we can spell it out more explicitly.

Reply:
Is this not what it's already saying?  895 contains the "MUST" requirement.  I don't want to call out HTTP; I'd prefer to stay more general and say that if the media type parameter is available in general.  It could come from anywhere, e.g. the OS or app.

Reply:
I guess I was confused what "Implementations which support media type parameter inputs" meant. Does that cover "if you used HTTP, you better check the Content-Type header and do what it says"?

Reply:
I think it covers any scenario, including http, where you might receive a media type.  So, yeah.